tools/data_source/upload.py

```diff
@@ -274,13 +274,18 @@ def add_file( dataset, registry, json_file, output_path ):
                     parts = dataset.name.split( "." )
                     if len( parts ) > 1:
                         ext = parts[-1].strip().lower()
-                        if not Binary.is_ext_unsniffable(ext):
-                            file_err( 'The uploaded binary file contains inappropriate content', dataset, json_file )
-                            return
-                        elif Binary.is_ext_unsniffable(ext) and dataset.file_type != ext:
-                            err_msg = "You must manually set the 'File Format' to '%s' when uploading %s files." % ( ext.capitalize(), ext )
-                            file_err( err_msg, dataset, json_file )
-                            return
+                        if not Binary.is_ext_unsniffable( ext ):
+                            format_information = Binary.is_sniffable_binary( dataset.path )
```

Comment:
`is_sniffable_binary()` seems to be called already at line 133, why is needed again here?